caravel/sql_lab.py

```diff
@@ -44,8 +45,9 @@ def create_table_as(sql, table_name, schema=None, override=False):
 @celery_app.task
 def get_sql_results(query_id, return_results=True):
     """Executes the sql query returns the results."""
-    db.session.commit()  # HACK
-    query = db.session.query(models.Query).filter_by(id=query_id).one()
+    session = db.session()
+    session.commit()  # HACK
```

Comment:
I believe you could do expunge(query) in the places you are looking it up using the db.session to make sure that the looked up object will be up to date.